### PR TITLE
Revert: Imaging pulling on delete

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: pkg/provider/pulumi/azure/azure.go [hunk 1]
```diff
@@ -195,12 +195,8 @@ func (a *azureProvider) Configure(ctx context.Context, autoStack *auto.Stack) er
 	return nil
 }
 
-func (a *azureProvider) ReadPulumiStack(ctx context.Context, backendURL string) (*common.PulumiStack, error) {
-	return nil, utils.NewNotSupportedErr("azure ReadPulumiStack()")
-}
-
-func (a *azureProvider) TryPullImage(ctx context.Context, imageURL string) error {
-	return utils.NewNotSupportedErr("azure TryPullImage()")
+func (a *azureProvider) TryPullImages() error {
```
Comment (asalkeld) [inline pkg/provider/pulumi/azure/azure.go]: can you just totally remove the TryPullImages?

File: pkg/provider/pulumi/gcp/gcp.go [hunk 2]
```diff
@@ -208,34 +206,18 @@ func (g *gcpProvider) Configure(ctx context.Context, autoStack *auto.Stack) erro
...
-		}
-
-		return nil, err
-	}
-
-	return common.StackFromReader(r)
-}
-
-func (g *gcpProvider) TryPullImage(ctx context.Context, imageURL string) error {
+func (g *gcpProvider) TryPullImages() error {
```
Comment (asalkeld) [inline pkg/provider/pulumi/gcp/gcp.go]: just remove all of this..

**Review by asalkeld — CHANGES_REQUESTED** [review]
can we just totally remove the tryPullImages?

**Review by jyecusch — APPROVED** [review]

File: pkg/provider/pulumi/azure/azure.go (hunk 1 above)
Comment (tjholm) [reply pkg/provider/pulumi/azure/azure.go]: @asalkeld there is still an implementation for TryPullImages for stack update that was implemented prior to these changes to try and fix ECR deletion.

Comment (tjholm) [reply pkg/provider/pulumi/azure/azure.go]: or were these implemented for the pro backend?

Comment (tjholm) [reply pkg/provider/pulumi/azure/azure.go]: removed

File: pkg/provider/pulumi/gcp/gcp.go (hunk 2 above)
Comment (tjholm) [reply pkg/provider/pulumi/gcp/gcp.go]: removed

**Review by asalkeld — APPROVED** [review]

**Pull request merged**